src/emitters/distant.cpp

```diff
@@ -0,0 +1,157 @@
+#include <mitsuba/core/bsphere.h>
+#include <mitsuba/core/plugin.h>
+#include <mitsuba/core/properties.h>
+#include <mitsuba/core/warp.h>
+#include <mitsuba/render/emitter.h>
+#include <mitsuba/render/scene.h>
+#include <mitsuba/render/texture.h>
+
+NAMESPACE_BEGIN(mitsuba)
+
+/**!
+
+.. _emitter-distant:
+
+Distant directional emitter (:monosp:`distant`)
+-----------------------------------------------
+
+.. pluginparameters::
+
+    * - irradiance
+      - |spectrum|
+      - Spectral irradiance, i.e. amount of spectral power per unit area 
+        received by a hypothetical surface normal to the specified direction.
+    * - to_world
+      - |transform|
+      - Emitter-to-world transformation matrix.
+    * - direction
+      - |vector|
+      - Alternative (and exclusive) to `to_world`. Direction towards which the
+        emitter is radiating in world coordinates.
+
+This emitter plugin implements a distant directional source which radiates a
+specified power per unit area along a fixed direction. By default, the emitter
+radiates in the direction of the positive Z axis, i.e. :math:`(0, 0, 1)`.
+
+*/
+
+template <typename Float, typename Spectrum>
+class DistantEmitter final : public Emitter<Float, Spectrum> {
+public:
+    MTS_IMPORT_BASE(Emitter, m_flags, m_world_transform)
+    MTS_IMPORT_TYPES(Scene, Texture)
+
+    DistantEmitter(const Properties &props) : Base(props) {
+        /* Until `set_scene` is called, we have no information
+           about the scene and default to the unit bounding sphere. */
+        m_bsphere = ScalarBoundingSphere3f(ScalarPoint3f(0.f), 1.f);
+
+        if (props.has_property("direction")) {
+            if (props.has_property("to_world"))
+                Throw("Only one of the parameters 'direction' and 'to_world' "
+                      "can be specified at the same time!'");
+
+            ScalarVector3f direction(normalize(props.vector3f("direction")));
+            auto [up, unused] = coordinate_system(direction);
+
+            m_world_transform =
+                new AnimatedTransform(ScalarTransform4f::look_at(
+                    ScalarPoint3f(0.0f), ScalarPoint3f(direction), up));
+        }
+
+        m_flags      = EmitterFlags::Infinite | EmitterFlags::DeltaDirection;
+        m_irradiance = props.texture<Texture>("irradiance", Texture::D65(1.f));
```

Comment:
Maybe we could change `irradiance` to `radiance` throughout this whole file for consistency with `area.cpp` and other light source?

Reply:
You're right, we're actually returning a radiance (not divided by `$\pi$`). I'll fix that.